common/src/main/java/com/wynntils/mc/mixin/ScreenMixin.java

```diff
@@ -77,15 +84,14 @@ private void renderTooltipPre(
             ItemStack itemStack,
             int mouseX2,
             int mouseY2) {
-        ItemTooltipRenderEvent e = EventFactory.onItemTooltipRenderPre(poseStack, itemStack, mouseX, mouseY);
+        ItemTooltipRenderEvent.Pre e = EventFactory.onItemTooltipRenderPre(
+                poseStack, itemStack, Collections.unmodifiableList(tooltips), mouseX, mouseY);
         if (e.isCanceled()) return;
-
-        instance.renderTooltip(
-                poseStack,
-                instance.getTooltipFromItem(e.getItemStack()),
-                e.getItemStack().getTooltipImage(),
-                e.getMouseX(),
-                e.getMouseY());
+        List<ClientTooltipComponent> clientTooltips = e.getTooltips().stream()
```

Comment:
I'm not happy with this. I never liked the `@Redirect` in the first place (P0ke might remember my opposition ;-)). And now we're replacing vanilla code with something that is not even pretending to do the same as the original code. 

I don't have any alternative design suggestions, and I can't say that this will definitely break. But the more we mess around with vanilla code, the more we risk:

1) incompatibilities with future minecraft versions
2) incompatibilities with other mods
3) crashing the game if our code misbehaves (or has unintended interactions with other mods, or pieces of vanilla we did not consider), which
4) can't be turned off by disabling a feature.

Maybe there's no other way. But we need to be very, very sure about that, and also be sure the benefit outweighs the risks.